crates/rome_js_analyze/src/analyzers/nursery/use_collapsed_else_if.rs

```diff
@@ -0,0 +1,152 @@
+use crate::JsRuleAction;
+use rome_analyze::{context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_syntax::{AnyJsStatement, JsBlockStatement, JsElseClause, JsIfStatement};
+use rome_rowan::{AstNode, AstNodeList, BatchMutationExt};
+
+declare_rule! {
+    /// Enforce using `else if` instead of `else { if ... }`.
+    ///
+    /// If an `if` statement is the only statement in the `else` block, it is often clearer to use an `else if` form.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lonely-if
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     } else {
+    ///         // ...
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else if (anotherCondition) {
+    ///     // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else if (anotherCondition) {
+    ///     // ...
+    /// } else {
+    ///     // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///     // ...
+    /// } else {
+    ///     if (anotherCondition) {
+    ///         // ...
+    ///     }
+    ///     doSomething();
+    /// }
+    /// ```
+    ///
+    pub(crate) UseCollapsedElseIf {
+        version: "next",
+        name: "useCollapsedElseIf",
+        recommended: false,
+    }
+}
+
+pub(crate) struct RuleState {
+    block_statement: JsBlockStatement,
+    if_statement: JsIfStatement,
+}
+
+impl Rule for UseCollapsedElseIf {
+    type Query = Ast<JsElseClause>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let else_clause = ctx.query();
+        let alternate = else_clause.alternate().ok()?;
+        let block_statement = alternate.as_js_block_statement()?;
```

Comment:
Maybe it is possible to avoid the later cloning by destructing:

```suggestion
        let AnyJsExpression::JsBlockStatement(block_statement) = alternate else { return None; };
```